### PR TITLE
Adding troubleshooting guide

### PR DESCRIPTION
**Pull request by yorugac**
This is initial version for troubleshooting guide, issue: #331 
It is expected to be iteratively improved in the future.

### CODE REVIEW

File: docs/troublehooting.md [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+Passing the above leads to parsing errors which can be seen in the logs of either initializer or runner Pod, e.g.:
+```bash
+time="2024-01-11T11:11:27Z" level=error msg="invalid argument \"product_id=\\\"Test\" for \"--tag\" flag: parse error on line 1, column 12: bare \" in non-quoted-field"
+```
+
+This is a standard problem with escaping the characters, and there's even an [issue](https://github.com/grafana/k6-operator/issues/211) that can be upvoted.
+
+### Initializer logs an error but it's not about tags
+
+Often, this happens because of lack of attention to the [highly recommended](#⚠️-highly-recommended-⚠️) step. One more command that can be tried here is to run the following:
```
Comment (javaducky) [inline docs/troublehooting.md:143]: > [highly recommended](#⚠️-highly-recommended-⚠️)

🎗️ Don't forget to update if the anchor changes for the warning.

File: docs/troublehooting.md [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+- read carefully the output in initializer Pod: is it logged by k6 process or by something else?
+  - :information_source: k6-operator expects initializer logs to contain only the output of `k6 inspect`. If there's any other log line present, then k6-operator will fail to parse it and the test will not start. ([issue](https://github.com/grafana/k6-operator/issues/193))
+- check events in initializer Job and Pod as they may contain another hint about what is wrong
+
+### Non-existent ServiceAccount
+
+ServiceAccount can be defined as `serviceAccountName` and `runner.serviceAccountName` in PrivateLoadZone and TestRun CRD respectfully. If the specified ServiceAccount does not exist, k6-operator will successfully create Jobs but corresponding Pods will fail to be deployed, and k6-operator will wait indefinitely for Pods to be `Ready`. This error can be best seen in the events of the Job:
+
+```bash
+kubectl describe job plz-test-154546-1
```
Comment (javaducky) [inline docs/troublehooting.md:161]:
```suggestion
kubectl describe job plz-test-xxxxxx-x
```

File: docs/troublehooting.md [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+...
+Events:
+  Warning  FailedCreate  57s (x4 over 2m7s)  job-controller  Error creating: pods "plz-test-154546-1-" is forbidden: error looking up service account plz-ns/plz-sa: serviceaccount "plz-sa" not found
```
Comment (javaducky) [inline docs/troublehooting.md:164]:
```suggestion
  Warning  FailedCreate  57s (x4 over 2m7s)  job-controller  Error creating: pods "plz-test-xxxxxx-x-" is forbidden: error looking up service account plz-ns/plz-sa: serviceaccount "plz-sa" not found
```
Also...not sure if the trailing "-" was originally a typo, or valid. 🤔

File: docs/troublehooting.md [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+How to fix: incorrect `serviceAccountName` must be corrected and TestRun or PrivateLoadZone resource must be re-deployed.
+
+### Non-existent `nodeSelector`
+
+`nodeSelector` can be defined as `nodeSelector` and `runner.nodeSelector` in PrivateLoadZone and TestRun CRD respectfully. 
+
+This case is very similar to [ServiceAccount one](#non-existent-serviceaccount): the Pod creation will fail, only the error would be somewhat different:
+
+```bash
+kubectl describe job plz-test-154546-1
```
Comment (javaducky) [inline docs/troublehooting.md:178]:
```suggestion
kubectl describe job plz-test-xxxxxx-x
```

**Review by javaducky — COMMENTED** [review]
Just minor changes that I noticed.

File: docs/troublehooting.md [hunk 5]
```diff
@@ -0,0 +1,242 @@
...
+    ```bash
+    kubectl get jobs -A
+    kubectl describe job mytest-initializer
+    ```
+1. If one of the Pods was deployed but finished with `Error`, it makes sense to check its logs:
+    ```bash
+    kubectl logs mytest-initializer-xxxxx
+    ```
+
+If the Pods seem to be working but not producing an expected result and there's not enough information in the logs of the Pods, it might make sense to turn on k6 [verbose option](https://k6.io/docs/using-k6/k6-options/reference/#options) in `TestRun` spec:
```
[the remark this answers is not in the document]
Comment (yorugac) [reply docs/troublehooting.md:39]: Old habits :sweat_smile:

File: docs/troublehooting.md (hunk 3 above)
Comment (yorugac) [reply docs/troublehooting.md:164]: Yes, actually. It's a funny thing but apparently job-controller outputs the prefix for future pod here, hence with hyphen.

File: docs/troublehooting.md (hunk 4 above)
Comment (yorugac) [reply docs/troublehooting.md:178]: This one should actually be the Pod description :facepalm: I copied a wrong command; fixing.

File: docs/troublehooting.md [hunk 6]
```diff
@@ -0,0 +1,242 @@
+# Troubleshooting
+
+Just as any Kubernetes application, k6-operator can get into error scenarios which are sometimes a result of misconfigured test or setup. This document is meant to help troubleshoot such scenarios quicker.
+
+## Common tricks
+
+## :warning: Highly recommended! :warning:
```
[the remark this answers is not in the document]
Comment (yorugac) [reply docs/troublehooting.md]: Surprisingly, it is rendered fine:
image

**Comment by yorugac** [comment]
Thanks for the review @javaducky! :raised_hands:

**Pull request merged**